apps/api/src/app/events/e2e/digest-events.e2e.ts

```diff
@@ -1046,4 +1044,122 @@ describe('Trigger event - Digest triggered events - /v1/events/trigger (POST)',
 
     expect(message[0].content).to.include('HAS_DIGEST_PROP');
   });
+
+  it('should merge digest events accordingly when concurrent calls', async () => {
+    template = await session.createTemplate({
+      steps: [
+        {
+          type: StepTypeEnum.DIGEST,
+          content: '',
+          metadata: {
+            unit: DigestUnitEnum.MINUTES,
+            amount: 5,
+            type: DigestTypeEnum.REGULAR,
+          },
+        },
+        {
+          type: StepTypeEnum.IN_APP,
+          content: 'Hello world {{step.events.length}}' as string,
+        },
+      ],
+    });
+
+    const result = await Promise.all([
+      triggerEvent({
+        customVar: 'concurrent-call-1',
+      }),
+      triggerEvent({
+        customVar: 'concurrent-call-2',
+      }),
+      triggerEvent({
+        customVar: 'concurrent-call-3',
+      }),
+      triggerEvent({
+        customVar: 'concurrent-call-4',
+      }),
+      triggerEvent({
+        customVar: 'concurrent-call-5',
+      }),
+    ]);
+
+    await session.awaitRunningJobs(template?._id, false, 5);
```

Comment:
We trigger 5 events so we would expect 5 running jobs, therefore the third param of this helper, `unfinishedJobs` is set to 5 to move on with the test execution.